packages/cli/src/config/index.ts

```diff
@@ -17,6 +18,9 @@ if (inE2ETests) {
 	process.env.N8N_ENCRYPTION_KEY = 'test_key';
 	process.env.N8N_PUBLIC_API_DISABLED = 'true';
 	process.env.SKIP_STATISTICS_EVENTS = 'true';
+
+	// teardown script defaults to the real user dir, so override it
+	if (!process.env.N8N_USER_FOLDER?.endsWith('/n8n-tests/')) setUpTempUserDirSync();
```

Comment:
Initially I wanted to set up this temp dir in the teardown script itself, but for some reason this had no effect.